tests/Unit/Adapter/Image/ProductImageValidatorTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Adapter\Image;
+
+use Generator;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use PrestaShop\PrestaShop\Adapter\Image\ImageValidator;
+use PrestaShop\PrestaShop\Adapter\Image\ProductImageValidator;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\UploadedProductImageConstraintException;
+use Tests\Resources\DummyFileUploader;
+
+class ProductImageValidatorTest extends TestCase
+{
+    /**
+     * @dataProvider getInvalidMaxUploadSizesForFile
+     */
+    public function testItThrowsExceptionWhenFileSizeIsLargerThanAdministrableMaxUploadSize(string $filePath, int $maxUploadSize)
+    {
+        $imageValidator = $this->createMock(ImageValidator::class);
+        $configurator = $this->createMock(Configuration::class);
+        $configurator->method('get')->with('PS_LIMIT_UPLOAD_IMAGE_VALUE')->willReturn($maxUploadSize);
+        $imageValidator = new ProductImageValidator($imageValidator, $configurator);
+        $this->expectException(UploadedProductImageConstraintException::class);
+
+        $imageValidator->assertFileUploadLimits($filePath);
+    }
+
+    public function getInvalidMaxUploadSizesForFile(): Generator
```

Comment:
You should also have a test where the exception is not thrown (just relying on this test we could throw the exception all the time and they would still be green 😛) Either via a dedicated test endpoint or by providing an additional boolean in this data provider to indicate if the exception is expected or not